Introduce support for Orange PI Zero 2

pblxptr opened the pull request:
This pull request introduces support for [Orange Pi Zero 2 board ](http://www.orangepi.org/html/hardWare/computerAndMicrocontrollers/details/Orange-Pi-Zero-2.html)

Kernel: Mainline 5.15.35
Device tree: taken from Kernel 6x (since Orange Pi Zero 2 barely, but, it is supported in the newest version) and adapted to Kernel 5x.
Ethernet: enabled & working
WIFI: enabled & working

nandra commented on conf/machine/orange-pi-zero2.conf:
```diff
@@ -0,0 +1,19 @@
+
+require conf/machine/include/sun50i.inc
+
+PREFERRED_PROVIDER_virtual/kernel = "linux-mainline"
```
we don't need this line, it's default

nandra commented on conf/machine/orange-pi-zero2.conf:
```diff
@@ -0,0 +1,19 @@
+UBOOT_MACHINE = "orangepi_zero2_defconfig"
+
+UBOOT_BINARY = "u-boot-nodtb.bin"
+SPL_BINARY = "u-boot-sunxi-with-spl.bin"
```
Those lines are also not necessary

pblxptr replied:
Hmm, I think that SPL_BINARY value being set here is actually needed - when I comment this out, then the board does not boot. 
I see that in sunxi.inc (not used by orange-pi-zero2.conf) this variable is set to the same value as here, however, in the sunx64 (which is included by sun50i.inc and then by orange-pi-zero2.conf) the SPL_BINARY variable has a different value. Why is that?

nandra replied:
Don't know why is that set in sunxi64. Then keep it set please.

nandra commented on conf/machine/orange-pi-zero2.conf:
```diff
@@ -0,0 +1,19 @@
+UBOOT_BINARY = "u-boot-nodtb.bin"
+SPL_BINARY = "u-boot-sunxi-with-spl.bin"
+
+MACHINEOVERRIDES:append = "sun50i:sun50i-h616:"
```
This is wrong. after :append the should be 'space'. also sun50i is not necessary as it's SOC_ARCH and added by default. MACHINEOVERRIDES usually added like that:
`MACHINEOVERRIDES =. "sun50i-h616:"`

pblxptr replied:
Have a look at the recent changes I've just committed. There is one thing that we have to solve here. For the purpose of setting an appropriate PLATFORM variable for ATF we have to distinguish between sun50i and sun50i-h616. Previously as you commented above, MACHINEOVERRIDES was added in the machine conf file overriding sun50i and sun50i-h616. It worked. But in the recent changes I've created a separate include file for sun50i-h616 (and there added MACHINEOVERRIDES) because as far as I know there is more than one board using this chip, so in my opinion it would be better to keep a separate include file for this SOC.

nandra commented on recipes-bsp/atf/atf-sunxi_git.bb:
```diff
@@ -1,18 +1,24 @@
 SRCREV = "aa75c8da415158a94b82a430b2b40000778e851f"
 
-SRC_URI:append = " file://0001-Use-same-type-as-in-declaration.patch"
+SRC_URI:sun50i-h616 = "git://github.com/ARM-software/arm-trusted-firmware;nobranch=1;protocol=https"
```
Would not make distinguish for h616 and other. I plan to move atf sunxi to mainline so it can be done now.

pblxptr replied:
So if I remove "apritzel" fork and keep only the mainline, will it work out-of-the-box for other sun50i-based boards or do we have to change something more?

nandra replied:
If you update it I'll try on  other board I have in stock. I think I've tried it once and it works so I didn't expect issues.

nandra commented on recipes-kernel/linux/linux-h616.bb:
```diff
@@ -0,0 +1,20 @@
+SECTION = "kernel"
```
This file is not necessary and can be dropped.

nandra commented on recipes-kernel/linux/linux-mainline_5.15.35.bb:
```diff
@@ -5,3 +5,16 @@ DESCRIPTION = "Mainline Longterm Linux kernel"
 
 SRC_URI[sha256sum] = "0a1a5ae2f30eb2b38215e59077f045aabd7f4e2857a881482f02ea48186105d8"
+
+SRC_URI:append:orange-pi-zero2sun50i = " file://defconfig"
```
Please change in way like:
```
SRC_URI:append:orange-pi-zero2 = " \
   file1 \
   file2 \
  etc
"```

nandra commented on recipes-kernel/uwe5622-firwmare/uwe5622-firmware.bb:
```diff
@@ -0,0 +1,20 @@
+
+COMPATIBLE_MACHINE = "orange-pi-zero2"
+
+SRC_URI += "file://wcnmodem.bin"
```
Same please combine to one SRC_URI sentence.

pblxptr commented:
> Please make mainline atf common for all 64 bit CPU's and we can merge it IMO ;). Thanks.

I've updated atf-sunxi recipe as you requested :)

nandra commented:
> > Please make mainline atf common for all 64 bit CPU's and we can merge it IMO ;). Thanks.
> 
> @nandra I've updated atf-sunxi recipe as you requested :)

I've tried today on A64 CPU but cannot make it working even with old atf on kirkstone. I remember it works on dunfell fine. Sorry to say that but please revert last commit and we can merge it as it is (don't want to add more things to debug ;)). Thanks.

pblxptr commented:
> > > Please make mainline atf common for all 64 bit CPU's and we can merge it IMO ;). Thanks.
> > 
> > 
> > @nandra I've updated atf-sunxi recipe as you requested :)
> 
> I've tried today on A64 CPU but cannot make it working even with old atf on kirkstone. I remember it works on dunfell fine. Sorry to say that but please revert last commit and we can merge it as it is (don't want to add more things to debug ;)). Thanks.

Reverted :)
